city_scrapers/spiders/il_corrections.py

```diff
@@ -0,0 +1,160 @@
+import re
+from datetime import datetime
+from io import BytesIO, StringIO
+
+import scrapy
+from city_scrapers_core.constants import ADVISORY_COMMITTEE
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from pdfminer.high_level import extract_text_to_fp
+from pdfminer.layout import LAParams
+
+
+class IlCorrectionsSpider(CityScrapersSpider):
+    name = "il_corrections"
+    agency = "Illinois Department of Corrections Advisory Board"
+    timezone = "America/Chicago"
+    start_urls = [
+        "https://www2.illinois.gov/idoc/aboutus/advisoryboard/Pages/default.aspx"
+    ]
+
+    def __init__(self):
+        self.links = {}
+        self.url_base = "https://www2.illinois.gov"
+        self.pdf_text = ""
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+        # Gather Meeting Dates/Links
+        self.links = self._parse_all_links(response)
+
+        for date in self.links.keys():
+            if "Agenda" in self.links[date].keys():
+                yield scrapy.Request(
+                    self.links[date]["Agenda"],
+                    callback=self._meeting,
+                    cb_kwargs=dict(date=date),
+                )
+            elif "Minutes" in self.links[date].keys():
+                yield scrapy.Request(
+                    self.links[date]["Minutes"],
+                    callback=self._meeting,
+                    cb_kwargs=dict(date=date),
+                )
+
+    def _meeting(self, response, date):
+        self.pdf_text = self._parse_pdf(response)
+        meeting = Meeting(
+            title=self._parse_title(),
+            description="",
+            classification=ADVISORY_COMMITTEE,
+            start=self._parse_times(date),
+            end=self._parse_times(date, False),
+            all_day=False,
+            time_notes="",
+            location=self._parse_location(),
+            links=self._parse_links(date),
+            source=response.url,
+        )
+
+        meeting["status"] = self._get_status(meeting)
```

Comment:
If cancellation is ever mentioned in meeting text, we can pass `pdf_text` to this as the `text` kwarg since we have some built-in handling of cancelling meetings based on text provided there

Reply:
Hmmm I don't think this would be the case? Since we're only parsing Agendas or Minutes they're for meetings that happened, I'm worried that the word cancel might be in the minute text for some other reason and falsely change the status. Though not a big change if we want to try it out!

Reply:
It's pretty rare that they include it, but not a big deal! We can leave it out